agent/auto-config/auto_config_test.go

```diff
@@ -1098,16 +1100,15 @@ func TestFallback(t *testing.T) {
 	// now wait for the fallback routine to be invoked
 	require.True(t, waitForChans(100*time.Millisecond, fallbackCtx.Done()), "fallback routines did not get invoked within the alotted time")
 
-	// persisting these to disk happens after the RPC we waited on above will have fired
-	// There is no deterministic way to know once its been written so we wrap this in a retry.
-	testretry.Run(t, func(r *testretry.R) {
-		resp, err := testAC.ac.readPersistedAutoConfig()
-		require.NoError(r, err)
+	testAC.stop()
+	<-testAC.ac.done
 
-		// ensure the roots got persisted to disk
-		require.Equal(r, thirdCert.CertPEM, resp.Certificate.GetCertPEM())
-		require.Equal(r, secondRoots.ActiveRootID, resp.CARoots.GetActiveRootID())
-	})
+	resp, err := testAC.ac.readPersistedAutoConfig()
```

Comment:
This was racing in `readPersistedAutoConfig` in a read to `AutoConfig.config.DataDir`, I believe because `AutoConfig.run` was still running and could have updated `ac.config`.

I removed the retry and ran the test ~150 times with no failures, so it seems the stop+wait makes it deterministic.

I see this pattern in other tests, but for whatever reason this one is the only one with a data race, so I only changed this one.

cc @mkeeler to make sure this test change is reasonable